CHANGELOG.md

```diff
@@ -30,6 +30,12 @@ Main (unreleased)
 - Read contextual attributes from Faro measurements (@codecapitano)
 - Rename Grafana Agent service in windows app and features to not include the description
 
+### Bugfixes
+
+- Restart managed components of a module loader only on if module content
+  changes or the last load failed. This was specifically impacting `module.git`
```

Comment:
Well, it impacts module.http too.

Reply:
Check my understanding: I don't think it is the case today that this happens for module.http or module.file since LoadFlowContent is called inside Update (means the module loader definition changed) and inside the OnStateChange of the embedded component which will only fire if the content is in fact updated.

https://github.com/grafana/agent/blob/33b7b61ed2421f3168cd9349a499201409e54284/component/module/http/http.go#L93

Reply:
Nope, you're right, my bad.